CONTRIBUTING.md

```diff
@@ -0,0 +1,586 @@
+- we use [CircleCI](https://circleci.com/gh/mapzen/vector-datasource) for continuous integration testing
+- we use [semver](http://semver.org/) for package versioning
+
+All  unit tests in a project will be automatically invoked when you commit to an existing project; make sure they exit successfully!
+
+## Active contributors
+
+We'll gladly invite active contributors to become members of the [Tilezen organization](https://github.com/Tilezen). New members will gain direct write permissions, *and with great power comes great responsibility*.
+
+## Project overview (what to change, where)
+
+Generally speaking there are three aspects of developing vector tiles.
+
+- Configuring project setup, see [wiki page](https://github.com/tilezen/vector-datasource/wiki/Mapzen-Vector-Tile-Service)
+- Updating database properties (can be done ahead of time or at runtime)
+- Changing how features are selected from the database (requires TileServer restart)
+
+<div class='alert-message'>
+Yellow call-outs like this are meant to draw your attention to an important idea or distinction you should keep in mind.
+</div>
+
+Map database in Postgres stores data from OpenStreetMap and other projects like Natural Earth and Who's On First.
+
+When data is loaded, database "triggers" calculate if a feature is included in which layer(s), at what "minimum zoom", and other Mapzen specific "mz" properties.
+
+When modifying the logic below, we'll need to update our Postgres functions, migrate the data, and cut new tiles.
+
+### Changing tile content in the vector-datasource repo
+
+Most content changes (eg: adding a new kind of feature) only require a database modification. Content changes are configured using YAML files which specify which database features are "filtered" and outputed in tiles. The location for these content filters is in the `yaml/` directory, for example: [pois.yaml](yaml/pois.yaml).
+
+Some preexisting feature filters are configured using an older raw SQL format in `jinja` files in the `queries/` directory. This older syntax is still helpful to select multiple feature properties across many kinds of features at once. For example, [pois.jinja](queries/pois.jinja).
+
+<div class='alert-message'>Generally perform maintenance on pre-existing jinja filters or optionally migrate them to the newer YAML format.</div>
+
+Additionally, the root `queries.yaml` specifies which `jinja` file to use per layer, and also specifies post-processing via Python transforms.
+
+### Serving tiles in the TileServer repo
+
+Listens for API requests on localhost, which are in the format of 0/0/0.ext
+
+When TileServer hears a request it asks Postgres for "the stuff" inside that tile's bounding box, configured via the `queries.yaml` file and `.jinja2` files.
+
+_**NOTE:** A change to one of the query files (jinja) requires TileServer to be restarted so they can be reloaded. But content filter changes (YAML) generally doesn't require restarting TileServer._
```

Comment:
Updates to the jinja don't require restarting tileserver; they are re-read on request during development. And the dev server will automatically detect any python changes and restart itself too. The only changes that require restarts are those to the configuration files themselves, like queries.yaml.

It's probably worth mentioning that changes to the layer yaml files will require at a minimum reloading the sql functions. This is sufficient if only the `kind` or any output properties have changed. For `min_zoom` updates, affected features will need to be recalculated, probably via a migration. But tileserver doesn't need to get restarted in these circumstances.